cpp/source/deal.II/supercapacitor.templates.h

```diff
@@ -443,16 +444,15 @@ void SuperCapacitor<dim>::load(const std::string &filename)
   _geometry->get_triangulation()->clear();
   _geometry->get_triangulation()->copy_triangulation(tmp);
 
-  dealii::types::boundary_id anode_boundary_id;
-  dealii::types::boundary_id cathode_boundary_id;
   std::shared_ptr<std::unordered_map<
-      std::string, std::vector<dealii::types::material_id>>> materials;
-  ia >> anode_boundary_id;
-  ia >> cathode_boundary_id;
+      std::string, std::set<dealii::types::material_id>>> materials;
+  std::shared_ptr<std::unordered_map<
+      std::string, std::set<dealii::types::boundary_id>>> boundaries;
   ia >> materials;
-  _geometry->set_anode_boundary_id(anode_boundary_id);
-  _geometry->set_cathode_boundary_id(cathode_boundary_id);
+  ia >> boundaries;
   _geometry->set_materials(materials);
+  _geometry->set_boundaries(boundaries);
+  // TODO: fill weight map
```

Comment:
I am not sure we want to load the weight map. This  prevents us to use a different partitioning strategy if we change the number of procs.